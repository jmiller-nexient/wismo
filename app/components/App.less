@import '../styles/themes/index.less';

.App {
    color: @dte-black;
    font-family: @dte-font-family;
    margin: 1rem;
}
