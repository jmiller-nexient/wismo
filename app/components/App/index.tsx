import * as React from 'react';
import { hot } from 'react-hot-loader';
import {Link, Route, Switch} from 'react-router-dom';
import Layout from 'components/Layout';
import TestPage from 'components/TestPage';
import NotFoundPage from 'components/NotFoundPage';

const App: React.FC = () => {
    return (
        <div>
            <nav>
              <Link to="/">Home</Link><br/>
              <Link to="/testpage">Test Page</Link><br />
            </nav>

            <Switch>
                <Route exact path="/">
                    <Layout />
                </Route>
                <Route path="/testpage">
                    <TestPage />
                </Route>
                <Route component={NotFoundPage} />
            </Switch>
        </div>
    );
};

export default hot(module)(App);
