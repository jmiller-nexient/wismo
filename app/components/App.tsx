import React, { Component } from 'react';
import { hot } from 'react-hot-loader';

import './App.less';

class App extends Component {
  render() {
    return (
      <div className="App">
        <h1>Hello, WISMO!</h1>
      </div>
    );
  }
}

export default hot(module)(App);
