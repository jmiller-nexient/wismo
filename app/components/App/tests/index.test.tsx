import React, { ReactElement } from 'react';
import renderer from 'react-test-renderer';

import App from '..';

describe('<App />', () => {
  it('should render correctly', () => {
    const tree: ReactElement = renderer.create(
      <App />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
