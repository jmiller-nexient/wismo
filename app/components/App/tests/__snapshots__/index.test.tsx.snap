// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<div>
  <div
    className="dte-home"
  >
    <h1>
      Hello, WISMO!
    </h1>
    <h2>
      This is the 
      local instance
    </h2>
    <div>
      <h4>
        Account Number component:
      </h4>
      <span
        className="dte-wismo-account-number"
      >
        9275 309 8675 3
      </span>
    </div>
  </div>
</div>
`;
